tests/Avalonia.Controls.UnitTests/ItemsControlTests.cs

```diff
@@ -62,6 +64,32 @@ public void Container_Should_Have_TemplatedParent_Set_To_Null()
         }
 
         [Fact]
+        public void Container_Child_Should_Have_LogicalParent_Set_To_Container()
+        {
+            using (UnitTestApplication.Start(TestServices.StyledWindow))
+            {
+                var root = new Window();
+                var target = new ItemsControl();
+
+                root.Content = target;
+
+                var templatedParent = new Button();
+                target.TemplatedParent = templatedParent;
+                target.Template = GetTemplate();
+
+                target.Items = new[] { "Foo" };
+
+                root.ApplyTemplate();
+                target.ApplyTemplate();
+                target.Presenter.ApplyTemplate();
+
+                var container = (ContentPresenter)target.Presenter.Panel.Children[0];
+
+                Assert.Equal(container.Child.Parent, container);
```

Comment:
The order for `Assert.Equals` arguments is `expected, actual` so these two args should be swapped.